pkg/api/types.go

```diff
@@ -266,6 +266,19 @@ type V20170131ARMContainerService struct {
 	*v20170131.ContainerService
 }
 
+// VlabsUpgradeContainerService is the type we read and write from file
+// needed because the json that is sent to ARM and acs-engine
+// is different from the json that the ACS RP Api gets from ARM
+type VlabsUpgradeContainerService struct {
+	TypeMeta
+	*vlabs.UpgradeContainerService
+}
+
+// UpgradeContainerService API model
+type UpgradeContainerService struct {
+	OrchestratorProfile *OrchestratorProfile `json:"orchestratorProfile,omitempty"`
```

Comment:
I thought about this. Properties are needed for PUT model to be ARM compliant. I couldn't find anything in ARM documentation (and also poked around in the CRP & Disk for e.gs) for other operations that follow this model.

We don't need fields like name, tag, location during non PUT operations. These are needed by ARM to distinguish between its fields vs RP's fields (everything inside properties that are just sent to the RPs without ARM parsing them).

E.g.:
https://docs.microsoft.com/en-us/rest/api/virtualmachinescalesets/manage-all-vms-in-a-set
https://docs.microsoft.com/en-us/rest/api/manageddisks/disks/disks-grant-access

So I couldn't think of a good reason to follow this format for the sake of it in upgrade. But happy to discuss if this is indeed needed..